bindings/experimental/distrdf/python/DistRDF/Backends/Base.py

```diff
@@ -157,18 +157,25 @@ def mapper(current_range):
             # environment
             initialization()
 
-            # Build rdf
-            start = int(current_range.start)
-            end = int(current_range.end)
-
             if treename is not None:
+                # Build TEntryList for this range:
+                elists = ROOT.TEntryList()
+
                 # Build TChain of files for this range:
                 chain = ROOT.TChain(treename)
-                for f in current_range.filelist:
-                    chain.Add(str(f))
+                for start, end, filename in zip(current_range.localstarts, current_range.localends,
+                                                current_range.filelist):
+                    elist = ROOT.TEntryList("", "", treename, filename)
+                    for entry in range(start, end):
+                        elist.Enter(entry)
+                    elists.Add(elist)
+                    chain.Add(filename)
```

Comment:
usually it's good to also pass the number of entries to `TChain::Add`, so TChain does not have to ask that information to the file later. I am not sure how that interacts with the TEntryList though. @pcanal could you comment?

Reply:
It should be orthogonal (the `TEntryList` part for a given sub-tree would only be 'in use' when the sub-tree is loaded).

Reply:
From [TChain::Add](https://root.cern/doc/master/classTChain.html#a78a896924ac6c7d3691b7e013bcbfb1c)
> if (nentries > 0, the file is not connected, nentries is assumed to be the number of entries in the file.

So in this case we would not be passing the number of entries in the file but the number of entries we will be reading from the file. Are we sure it won't bite us back afterwards?

Reply:
mmh no `TChain::Add` expects that you pass the number of entries in the tree, not the number of entries of the tree that you will read. I meant the former.

Reply:
Right so in order to do that we need to modify the logic that creates the `TreeRange` objects to also store the number of entries in each tree. I will make a commit to show the needed changes